File: agent/env_utils.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import os
from pathlib import Path
from typing import Optional

DASHSCOPE_BASE_URL = "https://dashscope.aliyuncs.com/compatible-mode/v1"

try:
    from dotenv import load_dotenv
except ImportError:
    load_dotenv = None


def load_project_env() -> None:
    if not load_dotenv:
        return

    agent_dir = Path(__file__).resolve().parent
    load_dotenv(agent_dir / ".env")
    load_dotenv(agent_dir.parent / ".env")


def get_env(name: str, default: Optional[str] = None) -> Optional[str]:
    load_project_env()
    return os.getenv(name, default)

File: node-agent/envUtils.js
const fs = require("fs");
const path = require("path");

const DASHSCOPE_BASE_URL = "https://dashscope.aliyuncs.com/compatible-mode/v1";

function parseEnvFile(filePath) {
  if (!fs.existsSync(filePath)) {
    return;
  }

  const content = fs.readFileSync(filePath, "utf8");
  for (const rawLine of content.split(/\r?\n/)) {
    const line = rawLine.trim();
    if (!line || line.startsWith("#")) {
      continue;
    }

    const separator = line.indexOf("=");
    if (separator < 0) {
      continue;
    }

    const key = line.slice(0, separator).trim();
    let value = line.slice(separator + 1).trim();
    if (
      (value.startsWith("\"") && value.endsWith("\"")) ||
      (value.startsWith("'") && value.endsWith("'"))
    ) {
      value = value.slice(1, -1);
    }

    if (key && process.env[key] === undefined) {
      process.env[key] = value;
    }
  }
}

function loadProjectEnv() {
  parseEnvFile(path.join(__dirname, ".env"));
  parseEnvFile(path.join(__dirname, "..", ".env"));
}

function getEnv(name, defaultValue = undefined) {
  loadProjectEnv();
  return process.env[name] || defaultValue;
}

module.exports = {
  DASHSCOPE_BASE_URL,
  getEnv,
  loadProjectEnv,
};
